R/discrepancies.R

```diff
@@ -14,6 +14,24 @@ build_discrep <- function(
     envir = environment(pbbo)
   )
 
+  if (!is.null(attr(internal_discrepancy_f, "name"))) {
+    if (attr(internal_discrepancy_f, "name") == "gaussian_kl_approx") {
+      res <- function(lambda) {
+        inner_val <- build_approx_kl_discrep_pop(
+          target_sampler = target_sampler,
+          prior_predictive_sampler = prior_predictive_sampler,
+          n_samples_for_approx = n_internal_prior_draws
+        )
+
+        return(inner_val(lambda))
```

Comment:
I don't understand why in the population setting this function needs to return `inner_val(lambda)` (i.e. the result), when in the covariate setting (below) it returns _just_ `inner_val`.

Reply:
Where is the missing/extra eval in each case?